src/main/java/graphql/normalized/ExecutableNormalizedOperationFactory.java

```diff
@@ -64,6 +65,50 @@
  */
 @PublicApi
 public class ExecutableNormalizedOperationFactory {
+    public static class Options {
+        private final GraphQLContext graphQLContext;
+        private final Locale locale;
+        private final int maxChildrenDepth;
+
+        private Options(GraphQLContext graphQLContext,
+                        Locale locale,
+                        int maxChildrenDepth) {
+            this.graphQLContext = graphQLContext;
+            this.locale = locale;
+            this.maxChildrenDepth = maxChildrenDepth;
+        }
+
+        public static Options defaultOptions() {
+            return new Options(
+                    GraphQLContext.getDefault(),
+                    Locale.getDefault(),
+                    Integer.MAX_VALUE);
+        }
+
+        public Options locale(Locale locale) {
+            return new Options(this.graphQLContext, locale, this.maxChildrenDepth);
+        }
+
+        public Options graphQLContext(GraphQLContext graphQLContext) {
+            return new Options(graphQLContext, this.locale, this.maxChildrenDepth);
+        }
+
+        public Options maxChildrenDepth(int maxChildrenDepth) {
+            return new Options(this.graphQLContext, this.locale, maxChildrenDepth);
+        }
+
+        public GraphQLContext getGraphQLContext() {
+            return graphQLContext;
+        }
+
+        public Locale getLocale() {
+            return locale;
+        }
+
+        public int getMaxChildrenDepth() {
+            return maxChildrenDepth;
+        }
+    }
```

Comment:
Nice one!

but can we have some javadoc on what they do please since its API

Reply:
Done 😄